opcua/server/history_sql.py

```diff
@@ -0,0 +1,165 @@
+from datetime import timedelta
+from datetime import datetime
+
+from opcua import ua, Node
+from opcua.server.history import HistoryStorageInterface
+
+import struct
+import sqlite3
+
+
+class HistorySQLite(HistoryStorageInterface):
+    """
+    very minimal history backend storing data in SQLite database
+    """
+    # FIXME: need to check on if sql_conn.commit() should be inside try block; and if .rollback() needs to be except
+
+    def __init__(self):
+        self._datachanges_period = {}
+        self._events = {}
+        self._db_file = "history.db"
+
+        self._conn = sqlite3.connect(self._db_file, detect_types=sqlite3.PARSE_DECLTYPES, check_same_thread=False)
+
+    def new_historized_node(self, node, period, count=0):
+        _c_new = self._conn.cursor()
+
+        node_id = self._get_table_name(node)
+
+        self._datachanges_period[node_id] = period
+
+        # create a table for the node which will store attributes of the DataValue object
+        try:
+            _c_new.execute('CREATE TABLE "{tn}" (ServerTimestamp TIMESTAMP,'
+                           ' SourceTimestamp TIMESTAMP,'
+                           ' StatusCode INTEGER,'
+                           ' Value TEXT,'
+                           ' VariantType INTEGER,'
+                           ' ValueBinary BLOB)'.format(tn=node_id))
+
+        except sqlite3.Error as e:
+            print(node_id, 'Historizing SQL Table Creation Error:', e)
+
+        self._conn.commit()
+
+    def save_node_value(self, node, datavalue):
+        _c_sub = self._conn.cursor()
+
+        node_id = self._get_table_name(node)
+
+        value_blob = self._pack_value(datavalue)
+
+        # insert the data change into the database
+        try:
+            _c_sub.execute('INSERT INTO "{tn}" VALUES (?, ?, ?, ?, ?, ?)'.format(tn=node_id), (datavalue.ServerTimestamp,
+                                                                                               datavalue.SourceTimestamp,
+                                                                                               datavalue.StatusCode.value,
+                                                                                               str(datavalue.Value.Value),
+                                                                                               datavalue.Value.VariantType.value,
```

Comment:
what about saving VariantType.name instead of value? I think that it does not impact much performance and make things more readable. We can always change it later. just a suggestion...